engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -423,7 +429,8 @@ namespace dmGameSystem
 
             for (uint32_t* i = begin; i != end; ++i)
             {
-                const SpriteComponent* component = (SpriteComponent*) buf[*i].m_UserData;
+                uint32_t component_index = (uint32_t)buf[*i].m_UserData;
+                const SpriteComponent* component = (const SpriteComponent*) &components[component_index];
```

Comment:
We now store the index with the render list entry, instead of the actual pointer.